app/common/vault_api.py

```diff
@@ -50,12 +54,35 @@ def create_role(self, cluster_name):
                                     mount_point=f"kubernetes-{cluster_name}",
                                     bound_service_account_names=service_account_name,
                                     bound_service_account_namespaces="*",
-                                    policies=[policy_name], ttl="1h")
+                                    policies=policies, ttl="1h")
             return role_name, 0
         except Exception as e:
             self.logger.info("Vault create_role exception is: {}".format(e))
             return str(e), 1
 
+    def setup_oidc(self, roles):
+        if not M2M_AUTH_ENABLED or len(roles) == 0:
+            return 0, 'ok'
+        oidc_key = f'{self.owner}-{self.repo}'
+        try:
+            self.__auth_client()
+            self.oidc_client.oidc_create_key(oidc_key)
+            self.logger.debug(f'oidc key created: {oidc_key}')
+
+            for role in roles:
+                # create oidc role
+                template = f'{{"permissions":"{role}"}}'
+                self.oidc_client.oidc_create_role(role, oidc_key, template)
+                self.logger.debug(f'oidc role created: "{role}". template: "{template}"')
+
+                # create policy
+                oidc_policy_name = f"{role}-id-token-policy"
```

Comment:
Let's renamed this prefix `id-token-policy`  to  `oidc-token-policy`. I will update issue description.